### PR TITLE
refactor(sdk): Switch to `BaseProvider` in entitlement.ts

### PR DESCRIPTION
**Pull request by shuhuiluo**
Updated multiple functions to use `ethers.providers.BaseProvider` instead of `StaticJsonRpcProvider` for broader compatibility. This change ensures a more flexible provider usage across different wallet operations.

### CODE REVIEW

File: scripts/deploy-contracts.sh
Comment (texuf) [inline scripts/deploy-contracts.sh:55]: Both of these lines look the same. Should there be an else, or different code for geth?

Comment (shuhuiluo) [reply scripts/deploy-contracts.sh:55]: I mimicked line 45-48. As far as I understand, the River chain is always deployed by Anvil. But the Base chain may be on Anvil or a local Geth.

Comment (texuf) [reply scripts/deploy-contracts.sh:55]: yes you're right. thank you.

File: packages/web3/src/entitlement.ts
Comment (texuf) [inline packages/web3/src/entitlement.ts:498]: I'm 100% for making this change to the BaseProvider so that a user can use whatever provider they want. 

I'm not excited about routing all of our calls through someone elses contract via an address that is updated via NPM

Can we deploy our own multi call contract and use that address with this library?

File: contracts/.env.localhost
Comment (giuseppecrj) [inline contracts/.env.localhost:27]: Not sure I'm happy saving this bytecode into environment variables, we have our own MultiCall.sol contract, I would rather use that instead of an external contract for this.

File: packages/web3/src/entitlement.ts
Comment (giuseppecrj) [reply packages/web3/src/entitlement.ts:498]: We have our own MultiCall contract for this https://github.com/river-build/river/blob/main/contracts/src/diamond/utils/multicall/Multicall.sol

**Review by giuseppecrj — CHANGES_REQUESTED** [review]

File: packages/web3/src/entitlement.ts
Comment (shuhuiluo) [reply packages/web3/src/entitlement.ts:498]: 0xsequence should be an established team. Unfortunately their [`MulticallProvider`](https://www.npmjs.com/package/@0xsequence/multicall) uses their own [`MultiCallUtils`](https://github.com/0xsequence/wallet-contracts/blob/master/contracts/modules/utils/MultiCallUtils.sol) which is deployed and verified [universally](https://blockscan.com/address/0xdbbFa3cB3B087B64F4ef5E3D20Dda2488AA244e6). We can deploy our own instance but it will still have to adhere to their interface.
If using viem multicall, the [`Multicall3`](https://www.multicall3.com/) contract is used.

Comment (texuf) [reply packages/web3/src/entitlement.ts:498]: @shuhuiluo are you still going to pull the StaticJsonRpcProvider ==> BaseProvider change out of this pr and submit it? I'd love to see that change get in.

Comment (shuhuiluo) [reply packages/web3/src/entitlement.ts:498]: Sure.

**Review by texuf — APPROVED** [review]
approved once fully merged of course.

**Review by giuseppecrj — APPROVED** [review]

**Pull request merged**
